src/server/pps/server/api_server.go

```diff
@@ -613,26 +613,50 @@ func (a *apiServer) ListJobSet(request *pps.ListJobSetRequest, serv pps.API_List
 	// Track the jobsets we've already processed
 	seen := map[string]struct{}{}
 
+	number := request.Number
+	if number == 0 {
+		number = math.MaxInt64
+	}
+	paginationMarker := request.PaginationMarker
+
 	// Return jobsets by the newest job in each set (which can be at a different
 	// timestamp due to triggers or deferred processing)
 	jobInfo := &pps.JobInfo{}
-	err := a.jobs.ReadOnly(serv.Context()).List(jobInfo, col.DefaultOptions(), func(string) error {
+	opts := &col.Options{Target: col.SortByCreateRevision, Order: col.SortDescend}
+	if request.Reverse {
+		opts.Order = col.SortAscend
+	}
+	err := a.jobs.ReadOnly(serv.Context()).List(jobInfo, opts, func(string) error {
+		if number == 0 {
+			return errutil.ErrBreak
+		}
 		if _, ok := seen[jobInfo.Job.ID]; ok {
 			return nil
 		}
 		seen[jobInfo.Job.ID] = struct{}{}
 
+		if paginationMarker != nil {
```

Comment:
I don't think the SQL row's create time is the same as the job created time. I will confirm that and update you here. Ideally, we should be able to query based on the jobInfo.Created time. We don't have this functionality at the moment since we just store the proto in postgres. Unwrapping the protos into columns is on the refactoring roadmap so this implementation should definitely be changed after that lands.